test/e2e/utils/TestControl.php

```diff
@@ -117,21 +136,119 @@ public function init_test_project($projectCode = null, $projectName = null, $own
+                    $config->fields[$customFieldName] = new LexConfigMultiText();
+                    $config->fields[$customFieldName]->inputSystems = new ArrayOf();
+                    if ($extraOptions['inputSystems']) {
+                        foreach ($extraOptions['inputSystems'] as $ws) {
+                            $config->fields[$customFieldName]->inputSystems->ensureValueExists($ws);
+                        }
+                    }
+            };
+            $label = str_replace($prefix, '', $customFieldName);
+            $config->fields[$customFieldName]->label = str_replace(' ', '_', $label);
+            $config->fields[$customFieldName]->hideIfEmpty = false;
+        }
+        // PHP copies objects by value, not reference, so now we have to write the config back
+        switch($parentField) {
+            case 'entry': $project->config->entry = $config; break;
+            case 'senses': $project->config->entry->fields[LexConfig::SENSES_LIST] = $config; break;
+            case 'examples': $project->config->entry->fields[LexConfig::SENSES_LIST]->fields[LexConfig::EXAMPLES_LIST] = $config; break;
+        }
+
+        // Now make the custom field visible in all views
+        foreach ($project->config->roleViews as $role => $roleView) {
+            if (!array_key_exists($customFieldName, $roleView->fields)) {
+                if ($customFieldType == 'MultiUnicode' || $customFieldType == 'MultiString') {
+                    $roleView->fields[$customFieldName] = new LexViewMultiTextFieldConfig();
+                } else {
+                    $roleView->fields[$customFieldName] = new LexViewFieldConfig();
+                }
+                $roleView->fields[$customFieldName]->show = true;
+            }
+        }
+        foreach ($project->config->userViews as $userId => $userView) {
+            if (!array_key_exists($customFieldName, $userView->fields)) {
+                if ($customFieldType == 'MultiUnicode' || $customFieldType == 'MultiString') {
+                    $userView->fields[$customFieldName] = new LexViewMultiTextFieldConfig();
+                } else {
+                    $userView->fields[$customFieldName] = new LexViewFieldConfig();
+                }
+                $userView->fields[$customFieldName]->show = true;
+            }
+        }
+
+        $project->write();
+
+        return $customFieldName;
+    }
+
+    public function add_lexical_entry(string $projectCode, array $data)
+    {
+        $project = ProjectModel::getByProjectCode($projectCode);
+        $entry = new LexEntryModel($project);
+        LexEntryDecoder::decode($entry, $data);
+        return $entry->write();
+    }
+
+    public function get_project_json(string $projectCode) {
+        $db = MongoStore::connect(DATABASE);
+        $project = $db->projects->findOne(['projectCode' => $projectCode]);
+        return $project;
+    }
+
+    public function new_method() {
+        return 'hello';
+    }
```

Comment:
Looks like a method we don't need in this PR :)

Reply:
Sure. Will remove.